src/lib/DutchDecayLib.sol

```diff
@@ -84,6 +102,47 @@ library DutchDecayLib {
         return startAmount + delta;
     }
 
+    /// @notice returns the linear interpolation between the two points
+    /// @param startPoint The start of the decay
+    /// @param endPoint The end of the decay
+    /// @param currentPoint The current position in the decay
+    /// @param startAmount The amount of the start of the decay
+    /// @param endAmount The amount of the end of the decay
+    /// @dev rounds in favor of the swapper based on input or output
+    function v3LinearDecay(
+        uint256 startPoint,
+        uint256 endPoint,
+        uint256 currentPoint,
+        int256 startAmount,
+        int256 endAmount,
+        bool isInput
+    ) internal pure returns (int256) {
+        if (currentPoint >= endPoint) {
+            return endAmount;
+        }
+        uint256 elapsed = currentPoint - startPoint;
+        uint256 duration = endPoint - startPoint;
+        int256 delta;
+        if (isInput) {
+            // Because startAmount + delta is subtracted from the original amount,
+            // we want to maximize startAmount + delta to favor the swapper
+            if (endAmount < startAmount) {
+                delta = -int256(uint256(startAmount - endAmount).mulDivDown(elapsed, duration));
+            } else {
+                delta = int256(uint256(endAmount - startAmount).mulDivUp(elapsed, duration));
+            }
+        } else {
+            // For outputs, we want to minimize startAmount + delta to favor the swapper
+            if (endAmount < startAmount) {
+                delta = -int256(uint256(startAmount - endAmount).mulDivUp(elapsed, duration));
+            } else {
+                delta = int256(uint256(endAmount - startAmount).mulDivDown(elapsed, duration));
+            }
+        }
```

Comment:
A suggestion I have for this is to condense the functions in this file into two distinct functions:
`linearDecayRoundingDown` and `linearDecayRoundingUp`, and the decay() functions for Input and Output, check endAmount < startAmount and call roudingDown/Up depending on that